### PR TITLE
Update STYLE-GUIDE.md

### PR DESCRIPTION
**Pull request by wora**
Introducing API versioning strategy.

@mattklein123 @douglas-reid @geeknoid

### CODE REVIEW

File: STYLE-GUIDE.md [hunk 1]
```diff
@@ -41,6 +41,24 @@ conventions:
 
 ## Versioning
 
+### API Versioning
+
+Isiot APIs should use a simple versioning strategy based on
+major versions and releases, such as `v1alpha`, `v2beta`, or
```
Comment (xiaolanz) [inline STYLE-GUIDE.md]: Do we plan to introduce Istio specific versioning? This doesn't seem align with k8s resource versioning. Breaking changes at minor versions are certainly allowed. I don't see why we cannot have v1alpha1 v1alpha2...

Comment (wora) [reply STYLE-GUIDE.md]: `v1alpha1` and `v1alpha2` are separate releases. Breaking changes are allowed between releases.

File: STYLE-GUIDE.md [hunk 2]
```diff
@@ -41,6 +41,26 @@ conventions:
...
+`v3`. Within each release, there should not be any breaking
+change to released features, such as changing the type of
+a field type, renaming a field, or changing a field number.
+Breaking changes are allowed between different releases,
+such as `v1alpha1` and `v1alpha2`.
+
+Deprecating a feature in an API release is allowed by following
+the deprecation process. Stable release should have at least
```
Comment (geeknoid) [inline STYLE-GUIDE.md]: The style guide should not contain the detail of the deprecation policy, that's a separate concern.

**Comment by wora** [comment]
Delete wording about deprecation process.

File: STYLE-GUIDE.md [hunk 3]
```diff
@@ -41,6 +41,25 @@ conventions:
 
 ## Versioning
 
+### API Versioning
+
+Isiot APIs should use a simple versioning strategy based on
```
Comment (frankbu) [inline STYLE-GUIDE.md]: Istio APIs

**Comment by wora** [comment]
I fixed the typo. Please approve and merge.

**Pull request merged**